### PR TITLE
feat(impl):[TRI-1145] verify bpn from token with the configured one, …

### PR DESCRIPTION
**Pull request by ds-ext-kmassalski**
Verify bpn from token with the configured one before accessing API endpoint

### CODE REVIEW

File: irs-api/src/main/java/org/eclipse/tractusx/irs/controllers/IrsController.java [hunk 1]
```diff
@@ -121,7 +123,7 @@ public class IrsController {
     @IrsTimer("registerjob")
     @PostMapping("/jobs")
     @ResponseStatus(HttpStatus.CREATED)
-    @PreAuthorize("hasAuthority('view_irs')")
+    @PreAuthorize("@authorizationService.verifyBpn() && hasAuthority('view_irs')")
```
Comment (mkanal) [inline irs-api/src/main/java/org/eclipse/tractusx/irs/controllers/IrsController.java:126]: I would change that to a method - bean direct in a annotation doesnt imrpove readability and is really hard to debug.

File: irs-api/src/main/java/org/eclipse/tractusx/irs/controllers/IrsController.java [hunk 2]
```diff
@@ -172,7 +174,7 @@ public JobHandle registerJobForGlobalAssetId(final @Valid @RequestBody RegisterJ
     })
     @IrsTimer("getjob")
     @GetMapping("/jobs/{id}")
-    @PreAuthorize("hasAuthority('view_irs')")
+    @PreAuthorize("@authorizationService.verifyBpn() && hasAuthority('view_irs')")
```
Comment (mkanal) [inline irs-api/src/main/java/org/eclipse/tractusx/irs/controllers/IrsController.java:177]: I would change that to a method - bean direct in a annotation doesnt imrpove readability and is really hard to debug.

File: irs-api/src/main/java/org/eclipse/tractusx/irs/controllers/IrsController.java [hunk 3]
```diff
@@ -216,7 +218,7 @@ public Jobs getJobById(
     })
     @IrsTimer("canceljob")
     @PutMapping("/jobs/{id}")
-    @PreAuthorize("hasAuthority('view_irs')")
+    @PreAuthorize("@authorizationService.verifyBpn() && hasAuthority('view_irs')")
```
Comment (mkanal) [inline irs-api/src/main/java/org/eclipse/tractusx/irs/controllers/IrsController.java:221]: I would change that to a method - bean direct in a annotation doesnt imrpove readability and is really hard to debug.

File: irs-api/src/main/java/org/eclipse/tractusx/irs/controllers/IrsController.java [hunk 4]
```diff
@@ -258,7 +260,7 @@ public Job cancelJobByJobId(
     @IrsTimer("getjobbystate")
     @GetMapping("/jobs")
     @PageableAsQueryParam
-    @PreAuthorize("hasAuthority('view_irs')")
+    @PreAuthorize("@authorizationService.verifyBpn() && hasAuthority('view_irs')")
```
Comment (mkanal) [inline irs-api/src/main/java/org/eclipse/tractusx/irs/controllers/IrsController.java:263]: e.g.@PreAuthorize( isAuth() )

isAuth(){
 return @authorizationService.verifyBpn() && hasAuthority('view_irs')
}

File: irs-api/src/main/java/org/eclipse/tractusx/irs/services/AuthorizationService.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+package org.eclipse.tractusx.irs.services;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * Verifies if token from JWT claim is equal to configured BPN
```
Comment (mkanal) [inline irs-api/src/main/java/org/eclipse/tractusx/irs/services/AuthorizationService.java:30]: BPN contains in JWT Token matches BPN under which IRS product is registed.

File: irs-api/src/main/java/org/eclipse/tractusx/irs/services/SecurityHelperService.java [hunk 6]
```diff
@@ -37,10 +37,16 @@ public final class SecurityHelperService {
     private static final String UNKNOWN = "Unknown";
 
     public String getClientIdClaim() {
-        return getClaimOrUnknown("clientId", getAuthenticationFromSecurityContext());
+        return getClaimOrUnknown("clientId");
```
Comment (mkanal) [inline irs-api/src/main/java/org/eclipse/tractusx/irs/services/SecurityHelperService.java:40]: does mehtod return Unknown as String ??

File: irs-api/src/main/java/org/eclipse/tractusx/irs/services/SecurityHelperService.java [hunk 7]
```diff
@@ -37,10 +37,16 @@ public final class SecurityHelperService {
...
     }
 
-    private String getClaimOrUnknown(final String claimName, final Authentication authentication) {
+    public String getBpnClaim() {
+        return getClaimOrUnknown("bpn");
```
Comment (mkanal) [inline irs-api/src/main/java/org/eclipse/tractusx/irs/services/SecurityHelperService.java:44]: extract as constant

File: irs-api/src/test/java/org/eclipse/tractusx/irs/services/AuthorizationServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;
+
+class AuthorizationServiceTest {
+
+    @Test
```
Comment (mkanal) [inline irs-api/src/test/java/org/eclipse/tractusx/irs/services/AuthorizationServiceTest.java:41]: Empty or Unknown case sis that tested?

File: irs-api/src/main/java/org/eclipse/tractusx/irs/controllers/IrsController.java [hunk 9]
```diff
@@ -294,7 +296,7 @@ public PageResult getJobsByState(
                                          }),
     })
     @GetMapping("/aspectmodels")
-    @PreAuthorize("hasAuthority('view_irs')")
+    @PreAuthorize("@authorizationService.verifyBpn() && hasAuthority('view_irs')")
```
Comment (mkanal) [inline irs-api/src/main/java/org/eclipse/tractusx/irs/controllers/IrsController.java:299]: same as above

**Review by mkanal — CHANGES_REQUESTED** [review]

**Review by ds-jhartmann — APPROVED** [review]
LGTM

File: irs-api/src/main/java/org/eclipse/tractusx/irs/controllers/IrsController.java (hunk 1 above)
Comment (mkanal) [reply irs-api/src/main/java/org/eclipse/tractusx/irs/controllers/IrsController.java:126]: This is only pseudo code. I do not think that this is readable.

**Review by mkanal — APPROVED** [review]

**Review by ds-jkreutzfeld — DISMISSED** [review]

**Review by ds-jkreutzfeld — APPROVED** [review]

**Pull request merged**